src/data/finsupp/basic.lean

```diff
@@ -280,6 +280,27 @@ lemma single_left_inj (h : b ≠ 0) :
   and_false, or_false, eq_self_iff_true, and_true] using H,
  λ H, by rw [H]⟩
 
+lemma single_support_ne_bot (i : α) (h : b ≠ 0) :
+  (single i b).support ≠ ⊥ :=
+begin
+  have : i ∈ (single i b).support := by simpa using h,
+  intro H,
+  simpa [H]
+end
+
+lemma single_support_disjoint (h : b ≠ 0) {i j : α} :
+  disjoint (single i b).support (single j b).support ↔ i ≠ j :=
```

Comment:
Is this statement also true?
```suggestion
lemma single_support_disjoint (ha : a ≠ 0) (hb : b ≠ 0) {i j : α} :
  disjoint (single i a).support (single j b).support ↔ i ≠ j :=
```

Reply:
No, because then you have to say `a \ne b`.

Reply:
I don't follow...

Reply:
You're absolutely right, I got my supports mixed up. Generalizing.